operator/v1/types.go

```diff
@@ -59,6 +63,23 @@ type OperatorSpec struct {
 	ObservedConfig runtime.RawExtension `json:"observedConfig"`
 }
 
+type LogLevel string
```

Comment:
I would hope these at least match some already defined levels like https://godoc.org/github.com/sirupsen/logrus#Level  so that users don't have to relearn these new log levels?

Reply:
> I would hope these at least match some already defined levels like https://godoc.org/github.com/sirupsen/logrus#Level so that users don't have to relearn these new log levels?

I think we want to intentionally avoid that.  Doing so will promote confusion if they don't map exactly and they only would for one.

I claim that Normal is obvious.  Debug and Trace are largely understood and AtAllCost has precedent in other languages.

Reply:
>  Doing so will promote confusion if they don't map exactly and they only would for one.

can you expand on this? Why wouldn't they match, the operator can choose the mapping

> I claim that Normal is obvious

Hmm, `Normal` is ambigous. Does is mean default/info ?

> AtAllCost has precedent in other languages.

Wouldn't `Trace` already imply that i'm debugging and i don't care at what cost?

Also IMO, log level should default to a level that most users will benefit from, with room above and below for advanced users.

Reply:
> Wouldn't `Trace` already imply that i'm debugging and i don't care at what cost?

kube logging has normal (about loglevel 2), debug (5) that doesn't include every request/response,  trace (6) that lists all requests/response pairs and codes, and traceEverything (8) that includes all request and response bodies.